### ISSUE TITLE
Graphql @auth rules - Fails on nested relationship data when using an groupsField of a ID Indexed attribute.

### ISSUE THREAD

**Issue by dklein1211** [aws-amplify/amplify-category-api#1878]
### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v20.3.0

### Amplify CLI Version

12.2.1

### What operating system are you using?

Windows WSL Ubuntu

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No

### Describe the bug

**Project Overview**
- I am building an onboarding website using React, NextJS, and Amplify. 
- The application uses cognito to manage users. I have a group called Admin and then each accounts id is a group in cognito. When a cognito user is created a custom attribute is added called accountID. In addition there is a dynamoDB table called AuthUser that is a mirror of the cognito user so that I can create a relationship between the Account table and the AuthUsers in a 1 to many relationship.
- The objective is to limit users who are not admins to only be able to query and mutate data based on their accountID.

**Issue**
While signed into a user that is not an Admin and the app queries a table with nested relationships the query fails.
 
 
**GraphQL Schema**
```
type Account @model @searchable @auth(rules: [
  {allow: groups, groups: ["Admin"], operations: [read, create, update, delete]},
  {allow: groups, groupsField: "id", operations: [read, update]}
  ]) {
  id: ID!
  account_name: String!
  products: AWSJSON
  authUsers: [AuthUser] @hasMany(indexName: "byAccount", fields: ["id"])
  locations: [Location] @hasMany(indexName: "byAccount", fields: ["id"])
  devices: [Device] @hasMany(indexName: "byAccount", fields: ["id"])
  users: [AccountUser] @hasMany(indexName: "byAccount", fields: ["id"])
  dids: [PhoneNumber] @hasMany(indexName: "byAccount", fields: ["id"])
  stage: String,
  locked: Boolean
}

type AuthUser @model @auth(rules: [
  {allow: groups, groups: ["Admin"], operations: [read, create, update, delete]}, 
  {allow: groups, groupsField:"accountID", operations: [read, update]}
  ]) {
  id: ID!
  accountID: ID! @index(name: "byAccount")
  last_name: String
  first_name: String
  email: AWSEmail
}
```

**GraphQL Query:**
```
query GetAccount($id: ID!) {
    getAccount(id: $id) {
        id
        account_name
        stage,
        locked,
        products,
        authUsers {
            items {
                id,
                email,
                first_name,
                last_name,
            }
            nextToken
        },
    }
}
```

### Expected behavior

**Troubleshooting 1**
- When logged in as a user that is in the group Admin everything works.
- When logged in as a user that is not an admin the query fails.

**Troubleshooting 2**
**- I started to remove attributes from the query statement and noticed that any nested attribute that has a relationship with another table would fail. If I remove all attributes that reference another table the query would work.**
- When logged in as a user I tried to query Auth User (not nested) and it returns I tested querying Account(nested via Auth User) and it fails. If I remove the nested data from the query it succeeds. 

Example of Failed Query
```
query GetAccount($id: ID!) {
    getAccount(id: $id) {
        id
        account_name
        stage,
        locked,
        products,
        authUsers {
            items {
                id,
                email,
                first_name,
                last_name,
            }
            nextToken
        },
    }
}
```

**Example of Successful Query**
```
query GetAccount($id: ID!) {
    getAccount(id: $id) {
        id
        account_name
        stage,
        locked,
        products,
    }
}
```

**The Error Message**
After successfully testing with an Admin group user I am 100% sure my schema and types are correct. I am curtain that the issue has something to do with the groupFields in Account when trying to query AuthUser. And if I just query AuthUser it works just fine.

### Reproduction steps

Reference my schema and try to query Account with a nested table AuthUsers whos groupField value is of type ID! and indexed.

### Project Identifier

bd5885134c410bd1c722a5e7e4f301d4

### Log output

Browser Error
<details>

```
next-dev.js:20 Error: One or more parameter values were invalid: Condition parameter type does not match schema type (Service: DynamoDb, Status Code: 400, Request ID: 73NAQRPLFNSKFITFSRH0G2MNRNVV4KQNSO5AEMVJF66Q9ASUAAJG)
    at Object.eval [as queryFn] (fetcher.ts:20:19)
window.console.error @ next-dev.js:20
onError @ query.mjs:294
reject @ retryer.mjs:63
eval @ retryer.mjs:122
Promise.catch (async)
run @ retryer.mjs:106
createRetryer @ retryer.mjs:149
fetch @ query.mjs:307
executeFetch @ queryObserver.mjs:168
setOptions @ queryObserver.mjs:86
eval @ useBaseQuery.mjs:42
commitHookEffectListMount @ react-dom.development.js:23150
commitPassiveMountOnFiber @ react-dom.development.js:24926
commitPassiveMountEffects_complete @ react-dom.development.js:24891
commitPassiveMountEffects_begin @ react-dom.development.js:24878
commitPassiveMountEffects @ react-dom.development.js:24866
flushPassiveEffectsImpl @ react-dom.development.js:27039
flushPassiveEffects @ react-dom.development.js:26984
eval @ react-dom.development.js:26769
workLoop @ scheduler.development.js:266
flushWork @ scheduler.development.js:239
performWorkUntilDeadline @ scheduler.development.js:533


```

</details>

CloudWatch API Error Log
<details>

```
{
    "logType": "ResponseMapping",
    "path": [
        "getAccount",
        "authUsers"
    ],
    "fieldName": "authUsers",
    "resolverArn": "arn:aws:appsync:us-west-2:527661105615:apis/***********/types/Account/resolvers/authUsers",
    "functionName": "AccountAuthUsersDataResolverFn",
    "requestId": "97b83bd4-f9ce-4a48-a87b-20d5e25e9d07",
    "context": {
        "arguments": {},
        "source": {
            "createdAt": "2023-02-17T21:28:32.044Z",
            "stage": "new",
            "__typename": "Account",
            "account_name": "Audian",
            "id": "0cced1b5-e4d1-4ecd-918b-5056486fb90e",
            "locked": false,
            "products": {
                "seats": {
                    "total": 12,
                    "available": 10,
                    "inService": 0
                },
                "softphones": {
                    "total": 2,
                    "available": 4,
                    "inService": 0
                },
                "faxboxes": {
                    "total": 1,
                    "available": 2,
                    "inService": 0
                }
            },
            "updatedAt": "2023-03-21T20:25:21.869Z"
        },
        "prev": {
            "result": {}
        },
        "stash": {
            "authRole": "arn:aws:sts::527661105615:assumed-role/***********-101447-authRole/CognitoIdentityCredentials",
            "conditions": [],
            "connectionAttributes": {
                "id": [
                    "0cced1b5-e4d1-4ecd-918b-5056486fb90e"
                ]
            },
            "fieldName": "authUsers",
            "hasAuth": true,
            "metadata": {
                "dataSourceType": "AMAZON_DYNAMODB",
                "apiId": "*********m3ve"
            },
            "tableName": "AuthUser-*******3ve-dev",
            "typeName": "Account",
            "unauthRole": "arn:aws:sts::527661105615:assumed-role**********-101447-unauthRole/CognitoIdentityCredentials"
        },
        "error": {
            "message": "One or more parameter values were invalid: Condition parameter type does not match schema type (Service: DynamoDb, Status Code: 400, Request ID: 2AIVH5RKMA2EMTQJ9673KND3O3VV4KQNSO5AEMVJF66Q9ASUAAJG)",
            "type": "DynamoDB:DynamoDbException"
        },
        "outErrors": []
    },
    "fieldInError": true,
    "errors": [
        "CustomTemplateException(message=One or more parameter values were invalid: Condition parameter type does not match schema type (Service: DynamoDb, Status Code: 400, Request ID: 2AIVH5RKMA2EMTQJ9673KND3O3VV4KQNSO5AEMVJF66Q9ASUAAJG), errorType=DynamoDB:DynamoDbException, data=null, errorInfo=null)"
    ],
    "parentType": "Account",
    "graphQLAPIId": "********m3ve",
    "functionArn": "arn:aws:appsync:us-west-2:527661105615:apis/*********3ve/functions/k*******76tkq"
}
```

</details>

### Before submitting, please confirm:

- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

**Comment by dklein1211** [comment]
Hello, so I tried using owner instead of group as seen below and this resolved my issue. I am still curious why groups doesn't work the way it should.

```
type Account @model @searchable @auth(rules: [
  {allow: groups, groups: ["Admin"], operations: [read, create, update, delete]},
  {allow: owner, ownerField:"id", identityClaim:"custom:group", operations: [read, update]}
  ]) {
  id: ID!
  account_name: String!
  products: AWSJSON
  authUsers: [AuthUser] @hasMany(indexName: "byAccount", fields: ["id"])
  locations: [Location] @hasMany(indexName: "byAccount", fields: ["id"])
  devices: [Device] @hasMany(indexName: "byAccount", fields: ["id"])
  users: [AccountUser] @hasMany(indexName: "byAccount", fields: ["id"])
  dids: [PhoneNumber] @hasMany(indexName: "byAccount", fields: ["id"])
  stage: String,
  locked: Boolean
}
```

**Comment by AnilMaktala** [comment]
Hey @dklein1211, We appreciate you bringing up the issue and sharing the effective workaround you've found. We will look into the problem from our end and keep you informed of our investigation progress.

**Comment by phani-srikar** [comment]
Hi @dklein1211 Good to know that you're now using owner based Auth since your setup with dynamic group Auth is not supported since the `groupField` is not of type `String` or `[String]`. Also, storing group names as primary keys is not scalable as you would then need to create different groups for each user. I've tested a couple of variations of this schema and did not run into this error. Closing this issue. Thanks.
